src/contracts/actions.ts

```diff
@@ -156,17 +191,28 @@ export const overrideContractActions = (client: GenLayerClient<SimulatorChain>)
       );
     }
 
-    const encodedData = encodeFunctionData({
-      abi: client.chain.consensusMainContract?.abi as any,
-      functionName: "addTransaction",
-      args: [
-        senderAccount.address,
-        recipient,
-        client.chain.defaultNumberOfInitialValidators,
-        consensusMaxRotations,
-        data,
-      ],
-    });
+    let encodedData: `0x${string}`;
+    if (appeal) {
```

Comment:
@kstroobants instead of having this `appeal` flag here, let's make the _sendTransaction method to receive the encodedData as parameter.
You can have another private method around here that does the encoding for the addTransaction call